lib/modules/dashboard.js

```diff
@@ -88,14 +89,20 @@ module.go = async () => {
 const initialTabID = location.hash.replace('#', '') || 'dashboardContents';
 let selectedTabMenuItem;
 
-export function addTab(tabID, tabName) {
+export function addTab(tabID, tabName, tabModule) {
 	const $tabPage = $('<div class="dashboardPane">').appendTo('#siteTable.linklisting').hide();
 
 	const tabMenuItem = CreateElement.tabMenuItem({
 		text: tabName,
 		className: 'res-dashboard-tab',
 		prepend: true,
 	});
+	// if a module with options has been provided then we automatically append the gear link
+	if (tabModule && tabModule.options) {
```

Comment:
I think this function should take `tabModuleId` instead of an entire module; then this logic isn't hidden and the caller can decide whether they want a gear link by (not) passing a moduleId.

Also, technically we could have a module with no options that could still be enabled/disabled, so we'd still want to show the gear.